layers.go

```diff
@@ -72,6 +73,28 @@ type Layer struct {
 	// is set before using it.
 	Created time.Time `json:"created,omitempty"`
 
+	// CompressedDigest is the digest of the blob that was last passed to
+	// ApplyDiff() or Put(), as it was presented to us.
+	CompressedDigest digest.Digest `json:"compressed-diff-digest,omitempty"`
```

Comment:
Does any layer deduplication for layers with the same `UncompressedDigest` happen? If it does, it could be useful to support storing several `CompressedDigest+CompressedSize` values for such a deduplicated layer.

(Either way, storing at least one will handle most of the cases and is clearly better than nothing.)

Longer-term, it could be useful to record some kind of “here is where the compressed version of the layer can be downloaded from” pointer for each `CompressedDigest` value, to allow reproducing the original exactly (and therefore to allow “pushing” an image without breaking the manifest+signatures, or to allow pushing without copying the layer back to the registry it came from). That’s not happening soon, I’m only mentioning this to make sure something like that will be possible—looking at this PR, it seems that extending the metadata in the future will work just fine.